In [1]:
import pandas as pd
import numpy as np

# Load the dataset
df = pd.read_csv("my_data (1).csv")

# Display the first 5 rows
df.head()

,name,city,gender,profession,age,cgpa,placed
0,moriarity,NaN,female,phd,28.0,5.94,1
1,moriarity,asgard,NaN,bachelor,50.0,8.55,0
2,holmes,NaN,female,masters,18.0,5.56,0
3,sam,NaN,male,bachelor,25.0,8.57,1
4,sam,NaN,male,bachelor,19.0,8.76,1


In [2]:
# Check the number of rows and columns
print("Number of rows:", df.shape[0])
print("Number of columns:", df.shape[1])

# Display column names
print("\nColumn names:")
print(df.columns.tolist())

# Display dataset information
print("\nDataset Information:")
df.info()

Number of rows: 1100
Number of columns: 7

Column names:
['name', 'city', 'gender', 'profession', 'age', 'cgpa', 'placed']

Dataset Information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1100 entries, 0 to 1099
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   name        1009 non-null   object 
 1   city        905 non-null    object 
 2   gender      994 non-null    object 
 3   profession  921 non-null    object 
 4   age         982 non-null    float64
 5   cgpa        962 non-null    float64
 6   placed      1100 non-null   int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 60.3+ KB


In [3]:
# Check missing values in each column

missing_values = df.isnull().sum()

print("Missing values in each column:")
print(missing_values)

Missing values in each column:
name           91
city          195
gender        106
profession    179
age           118
cgpa          138
placed          0
dtype: int64


In [4]:
# Check for duplicate rows

duplicate_count = df.duplicated().sum()

print("Number of duplicate rows:", duplicate_count)

Number of duplicate rows: 65


In [5]:
# Remove duplicate rows

df = df.drop_duplicates()

print("Duplicate rows removed successfully.")
print("New number of rows:", df.shape[0])

Duplicate rows removed successfully.
New number of rows: 1035


In [6]:
# Check missing values before cleaning

print("Missing values before cleaning:")
print(df.isnull().sum())

Missing values before cleaning:
name           85
city          184
gender         99
profession    169
age           112
cgpa          132
placed          0
dtype: int64


In [7]:
# Handle missing values

# Fill missing values in text columns with the mode
text_columns = ["name", "city", "gender", "profession"]

for column in text_columns:
    df[column] = df[column].fillna(df[column].mode()[0])

# Fill missing values in numeric columns with the median
numeric_columns = ["age", "cgpa"]

for column in numeric_columns:
    df[column] = df[column].fillna(df[column].median())

print("Missing values handled successfully.")

Missing values handled successfully.


In [8]:
# Verify missing values after cleaning

print("Missing values after cleaning:")
print(df.isnull().sum())

Missing values after cleaning:
name          0
city          0
gender        0
profession    0
age           0
cgpa          0
placed        0
dtype: int64


In [9]:
# Calculate basic statistics for numeric columns

print("Mean values:")
print(df[["age", "cgpa"]].mean())

print("\nMedian values:")
print(df[["age", "cgpa"]].median())

Mean values:
age     28.370048
cgpa     7.606464
dtype: float64

Median values:
age     26.00
cgpa     7.89
dtype: float64


In [10]:
# Display value counts for categorical columns

print("Gender value counts:")
print(df["gender"].value_counts())

print("\nProfession value counts:")
print(df["profession"].value_counts())

print("\nPlaced value counts:")
print(df["placed"].value_counts())

Gender value counts:
gender
male      814
female    221
Name: count, dtype: int64

Profession value counts:
profession
bachelor    659
masters     256
phd         120
Name: count, dtype: int64

Placed value counts:
placed
0    521
1    514
Name: count, dtype: int64


In [11]:
# Final verification of the cleaned dataset

print("Final dataset shape:")
print(df.shape)

print("\nMissing values:")
print(df.isnull().sum())

print("\nDuplicate rows:")
print(df.duplicated().sum())

print("\nFirst 5 rows of cleaned dataset:")
display(df.head())

Final dataset shape:
(1035, 7)

Missing values:
name          0
city          0
gender        0
profession    0
age           0
cgpa          0
placed        0
dtype: int64

Duplicate rows:
3

First 5 rows of cleaned dataset:


,name,city,gender,profession,age,cgpa,placed
0,moriarity,wakanda,female,phd,28.0,5.94,1
1,moriarity,asgard,male,bachelor,50.0,8.55,0
2,holmes,wakanda,female,masters,18.0,5.56,0
3,sam,wakanda,male,bachelor,25.0,8.57,1
4,sam,wakanda,male,bachelor,19.0,8.76,1


In [12]:
# Remove duplicates created after filling missing values

before = df.shape[0]

df = df.drop_duplicates()

after = df.shape[0]

print("Duplicate rows removed:", before - after)
print("Final number of rows:", after)

Duplicate rows removed: 3
Final number of rows: 1032


In [13]:
# Save the cleaned dataset as a new CSV file

df.to_csv("cleaned_data.csv", index=False)

print("Cleaned dataset saved successfully as cleaned_data.csv")

Cleaned dataset saved successfully as cleaned_data.csv


## Data Cleaning & Exploration Report

### Dataset Overview
The dataset was loaded using Pandas and examined for its structure, data types, missing values, and duplicate records.

### Cleaning Steps Applied
1. Loaded the CSV dataset using Pandas.
2. Checked the number of rows and columns.
3. Checked column names and data types.
4. Identified missing values in the dataset.
5. Filled missing values in categorical columns using the mode.
6. Filled missing values in numerical columns using the median.
7. Identified and removed duplicate rows.
8. Calculated basic statistics including mean and median for numerical columns.
9. Analyzed value counts for categorical columns.
10. Verified the cleaned dataset and saved it as `cleaned_data.csv`.

### Final Result
The final cleaned dataset contains 1032 rows with no missing values and no duplicate rows.